### DISCUSSION TITLE
Title Case Selected Text

### DISCUSSION THREAD

**Discussion by dealingwith** [johnlindquist/kit#335] [Share]
Here's a really quick one I decided to spin up this morning:

[Install title-case](https://scriptkit.com/api/new?name=title-case&url=https://gist.githubusercontent.com/dealingwith/02ad42f0b334ed36faf7d034f8d97c76/raw/14467032b86ce587d70032f4cd29b913ef30234c/title-case.js)

```js
let { titleCase } = await npm("title-case");
let selectedText = await getSelectedText();
let value = titleCase(selectedText);
await setSelectedText(value);
```

...also, a tiny variant on the above I like to call The Proper Namifier™

```js
let { titleCase } = await npm("title-case");
let selectedText = await getSelectedText();
let value = titleCase(selectedText) + '™';
await setSelectedText(value);
```

**Comment by johnlindquist** [comment]
Lol, great idea™ !

**Comment by zacjones93** [comment]
My `change-case` script is very similar! 

```js
let { titleCase, sentenceCase, paramCase, camelCase } = await npm("text-case");

let text = await kit.getSelectedText();

let options = {
  TitleCase: titleCase,
  SentanceCase: sentenceCase,
  "param-case": paramCase,
  camelCase: camelCase,
};

let option = await arg(
  "What case do you want to convert to?",
  Object.keys(options)
);

let formattedText = options[option](text);

await kit.setSelectedText(formattedText);
```

**Reply by dealingwith** [reply]
@zacjones93 "SentanceCase" 😉

**Reply by dealingwith** [reply]
I quite like this one. I did update the options to be more readable: 

```javascript
let options = {
  "Title Case": titleCase,
  "Sentence case": sentenceCase,
  "param-case": paramCase,
  "camelCase": camelCase,
};
```

**Comment by brandonpittman** [comment]
Updated it to use `title`.

```js
let title = await npm("title");
let selectedText = await getSelectedText();
let value = title(selectedText);
await setSelectedText(value);
```

**Comment by lsminter** [comment]
I was having issues running this script. The error I was given was: `ReferenceError: kit is not defined`. I ended up having to import kit `let { kit } = await npm("@johnlindquist/kit");` and removed `kit` from `kit.setSelectedText` and `kit.getSelectedText` to make it work. Is there a better way of doing this?

**Reply by lsminter** [reply]
Just removed `kit` and the import and it works perfectly fine.
